grype/presenter/template/presenter.go

```diff
@@ -84,5 +85,19 @@ var funcMap = func() template.FuncMap {
 
 		return 0
 	}
+	f["byName"] = func(collection interface{}) interface{} {
+		t := reflect.TypeOf(collection)
+		if t.Kind() != reflect.Slice {
+			return collection
+		}
```

Comment:
Is this necessary, given the next check?

Reply:
I think you need this check before doing the next check since if it was NOT a slice then the next check would panic

Reply:
It won't panic because of the form `var, ok := thing.(cast)`